lib/widgets/components/tickets/tickets_section.dart

```diff
@@ -14,64 +13,59 @@ class TicketSection extends StatelessWidget {
+              // TODO consider using a nicer loading indicator
+              LoadingOverlay.of(context).show();
+            } else if (state is TicketUsed) {
+              final envState = context.read<EnvironmentCubit>().state;
+              LoadingOverlay.of(context).hide();
+              ReceiptOverlay.of(context).show(
+                receipt: state.receipt,
+                isTestEnvironment:
+                    envState is EnvironmentLoaded && envState.isTestEnvironment,
+              );
+            }
+          },
+          builder: (context, state) {
+            if (state is TicketsLoading) {
+              return const Text('loading');
+            } else if (state is TicketsLoaded) {
+              // States extending this are also caught on this
+              if (state.tickets.isEmpty) {
+                return const CoffeeCardPlaceholder();
               }
-            },
-            builder: (context, state) {
-              if (state is TicketsLoading) {
-                final ticketsCubit = context.read<TicketsCubit>();
-                ticketsCubit.getTickets();
-                return const Text('loading');
-              } else if (state is TicketsLoaded) {
-                // States extending this are also caught on this
-                if (state.tickets.isEmpty) {
-                  return const CoffeeCardPlaceholder();
-                }
-                return Column(
-                  children: state.tickets
-                      .map(
-                        (p) => Padding(
-                          padding: const EdgeInsets.only(bottom: 12.0),
-                          child: Hero(
-                            tag: p.productId,
-                            child: CoffeeCard(
-                              title: p.productName,
-                              amountOwned: p.count,
-                              productId: p.productId,
-                            ),
+              return Column(
+                children: state.tickets
+                    .map(
+                      (p) => Padding(
+                        padding: const EdgeInsets.only(bottom: 12.0),
+                        child: Hero(
+                          tag: p.productId,
+                          child: CoffeeCard(
+                            title: p.productName,
+                            amountOwned: p.count,
+                            productId: p.productId,
                           ),
                         ),
-                      )
-                      .toList(),
-                );
-              } else if (state is TicketsError) {
-                //FIXME: display error
-                return const Text('error');
-              }
+                      ),
+                    )
+                    .toList(),
+              );
+            } else if (state is TicketsError) {
+              //FIXME: display error
+              return const Text('error');
```

Comment:
We should show something more descriptive than this :P

Reply:
Check latest commit